### PR TITLE
[CE] An NPE is thrown if port 1883 is in use when HiveMQ CE is started

### PR DESCRIPTION
**Pull request by tgracchus**
**Motivation**

Resolves https://github.com/hivemq/hivemq-community-edition/issues/403
and
https://hivemq.kanbanize.com/ctrl_board/42/cards/16936/details/

**Changes**

### CODE REVIEW

File: src/test/java/com/hivemq/bootstrap/StartupListenerVerifierTest.java [hunk 1]
```diff
@@ -15,48 +15,47 @@
...
     @Test(expected = UnrecoverableException.class)
     public void test_verifier_verify_only_listener_failed() throws Exception {
-        final ListenerStartupInformation failed = ListenerStartupInformation.failedListenerStartup(2000,
-                new TcpListener(2000, "0.0.0.0"),
-                new RuntimeException("reason"));
-        new StartupListenerVerifier(Lists.newArrayList(failed)).verifyAndPrint();
+        final ListenerStartupInformation failed =
+                ListenerStartupInformation.failedListenerStartup(new TcpListener(2000, "0.0.0.0"),
+                        new RuntimeException("reason"));
+        new StartupListenerVerifier(List.of(failed)).verifyAndPrint();
```
Comment (mchernyakov) [inline src/test/java/com/hivemq/bootstrap/StartupListenerVerifierTest.java:35]: is there any version for a single element? we could avoid allocating a list

Comment (tgracchus) [reply src/test/java/com/hivemq/bootstrap/StartupListenerVerifierTest.java:35]: you mean the constructor or the class itself?

Comment (mchernyakov) [reply src/test/java/com/hivemq/bootstrap/StartupListenerVerifierTest.java:35]: the constructor, like `new StartupListenerVerifier(failed)`

**Review by mchernyakov — APPROVED** [review]

File: src/test/java/com/hivemq/bootstrap/StartupListenerVerifierTest.java (hunk 1 above)
Comment (mchernyakov) [reply src/test/java/com/hivemq/bootstrap/StartupListenerVerifierTest.java:35]: in that case, yes. LGTM

File: src/main/java/com/hivemq/bootstrap/HiveMQNettyBootstrap.java [hunk 2]
```diff
@@ -291,14 +291,12 @@ private static class UpdateGivenFutureListener implements ChannelFutureListener
         @Override
         public void operationComplete(final @NotNull ChannelFuture future) throws Exception {
```
Comment (hlohse) [inline src/main/java/com/hivemq/bootstrap/HiveMQNettyBootstrap.java:292]: Noticed "Exception 'java.lang.Exception' is never thrown in the method" in IntelliJ:
```suggestion
        public void operationComplete(final @NotNull ChannelFuture future) {
```

**Pull request merged**